server/src/main/java/org/elasticsearch/http/AbstractHttpServerTransport.java

```diff
@@ -291,7 +314,22 @@ protected void serverAcceptedChannel(HttpChannel httpChannel) {
         boolean addedOnThisCall = httpChannels.add(httpChannel);
         assert addedOnThisCall : "Channel should only be added to http channel set once";
         totalChannelsAccepted.incrementAndGet();
-        httpChannel.addCloseListener(ActionListener.wrap(() -> httpChannels.remove(httpChannel)));
+        httpChannelStats.put(
+            HttpStats.ClientStats.getChannelKey(httpChannel),
+            new HttpStats.ClientStats(threadPool.absoluteTimeInMillis())
+        );
+        httpChannel.addCloseListener(ActionListener.wrap(() -> {
+            try {
+                httpChannels.remove(httpChannel);
+                HttpStats.ClientStats clientStats = httpChannelStats.get(HttpStats.ClientStats.getChannelKey(httpChannel));
+                if (clientStats != null) {
+                    clientStats.closedTimeMillis = threadPool.absoluteTimeInMillis();
+                }
+            } catch (Exception e) {
+
```

Comment:
Can you add a comment and maybe trace logging here, just in case we need it in the future? I don't think we should swallow exceptions without ever making them apparent elsewhere.